src/mraa.c

```diff
@@ -116,6 +123,31 @@ imraa_init()
            mraa_get_version(), (proc_user != NULL) ? proc_user->pw_name : "<unknown>", proc_euid);
 
     mraa_platform_t platform_type;
+
+    // Check to see if the enviroment variable has been set
+    env_var = getenv(JSONPLAT_ENV_VAR);
+    if (env_var != NULL) {
+        // Check to see if it's a file
+        if (access(env_var, R_OK) != -1) {
```

Comment:
asking for permission is pointless since you don't know file changed/detelted etc...